### PR TITLE
fix(DomRegistry): add missing elements

### LINKED ISSUE #11219: bug(DomSchema): `<time>`, `<summary>` els aren't allowed
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
`<time>`, `<summary>` element isn't allowed.

**Expected/desired behavior**
The tag is a standard HTML5 tag.
https://www.w3.org/TR/html5/text-level-semantics.html#the-time-element
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/time
It should be recognized correctly.

**Reproduction of the problem**
http://plnkr.co/edit/SZJms0

**What is the expected behavior?**
broken. (please see console log)

```
'time' is not a known element:
1. If 'time' is an Angular component, then verify that it is part of this module.
2. If 'time' is a Web Component then add "CUSTOM_ELEMENTS_SCHEMA" to the '@NgModule.schema' of this component to suppress this message. ("
    <div>
      <h2>Hello {{name}}</h2>
      [ERROR ->]<time>20:00</time>
    </div>
  "): App@3:6 ; Zone: <root> ; Task: Promise.then ; Value: Error: Template parse errors:
```

**What is the motivation / use case for changing the behavior?**

**Please tell us about your environment:**

* **Angular version:** 2.0.0-rc.6
 
* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ] 
 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]

### PR DESCRIPTION
**Pull request by vicb**
fixes #11219 
- The first commit improves readability - important for security sensitive code,
- The code commit add 4 missing tags (data, menuitem, summary, tags)

@mhevery @mprobst @rjamet Could you please double check the security impact ?

Might be nice to have in Angular2 final (it was detected the day RC6 was announced, some devs out there probably use some of those tags)

/cc @laco0416 Thanks for the initial report !

### CODE REVIEW

**Comment by mhevery** [comment]
Lgtm

File: modules/@angular/compiler/src/schema/dom_element_schema_registry.ts [hunk 1]
```diff
@@ -71,161 +72,167 @@ const OBJECT = 'object';
...
+          ':svg:tspan^:svg:textPositioning|',
+          ':svg:text^:svg:textPositioning|',
+          ':svg:textPath^:svg:textContent|',
+          ':svg:title^:svg:|',
+          ':svg:use^:svg:graphics|',
+          ':svg:view^:svg:|#zoomAndPan',
+          'data^[HTMLElement]|value',
+          'menuitem^[HTMLElement]|type,label,icon,!disabled,!checked,radiogroup,!default',
+          'summary^[HTMLElement]|',
+          'time^[HTMLElement]|dateTime',
```
Comment (laco0416) [inline modules/@angular/compiler/src/schema/dom_element_schema_registry.ts]: Isn't `HTMLTimeElement` ?

Comment (laco0416) [inline modules/@angular/compiler/src/schema/dom_element_schema_registry.ts]: And I think `dataTime` is IDL attribute name. As HTML attribute , `datetime` is correct. Is this wrong?
https://www.w3.org/TR/html5/text-level-semantics.html#the-time-element

File: modules/@angular/compiler/test/schema/dom_element_schema_registry_spec.ts [hunk 2]
```diff
@@ -46,9 +54,19 @@ export function main() {
...
     });
 
+    // https://github.com/angular/angular/issues/11219
+    it('should detect properties on elements missing from Chrome', () => {
+      expect(registry.hasProperty('data', 'value', [])).toBeTruthy();
+
+      expect(registry.hasProperty('menuitem', 'type', [])).toBeTruthy();
+      expect(registry.hasProperty('menuitem', 'default', [])).toBeTruthy();
+
+      expect(registry.hasProperty('time', 'dateTime', [])).toBeTruthy();
```
Comment (laco0416) [inline modules/@angular/compiler/test/schema/dom_element_schema_registry_spec.ts]: ditto

File: modules/@angular/compiler/test/schema/schema_extractor.ts [hunk 3]
```diff
@@ -6,10 +6,37 @@
...
+
+// Elements missing from Chrome (HtmlUnknownElement), to be manually added
+const MISSING_FROM_CHROME: {[el: string]: string[]} = {
+  'data^[HTMLElement]': ['value'],
+  // TODO(vicb): Figure out why Chrome and WhatWG do not agree on the props
+  // 'menu^[HTMLElement]': ['type', 'label'],
+  'menuitem^[HTMLElement]':
+      ['type', 'label', 'icon', '!disabled', '!checked', 'radiogroup', '!default'],
+  'summary^[HTMLElement]': [],
+  'time^[HTMLElement]': ['dateTime'],
```
Comment (laco0416) [inline modules/@angular/compiler/test/schema/schema_extractor.ts]: ditto

**Comment by rjamet** [comment]
LGTM too, these don't look that sensitive.

File: modules/@angular/compiler/src/schema/dom_element_schema_registry.ts (hunk 1 above)
Comment (vicb) [inline modules/@angular/compiler/src/schema/dom_element_schema_registry.ts]: Those are properties, not attributes

**Comment by mprobst** [comment]
Could you rebase onto master and push? We need an intermediate change to test this well internally.

**Comment by mprobst** [comment]
Merged as d309f7799c12a0b6ac70b64552da2bd2ee9ff872.

**Pull request closed without merging**